mmtk/src/scan_boot_image.rs

```diff
@@ -121,3 +134,18 @@ fn decode_long_encoding(cursor: Address) -> usize {
         value
     }
 }
+
+
+pub struct ScanBootImageRoots<E: ProcessEdgesWork<VM=JikesRVM>>(usize, usize, PhantomData<E>);
+
+impl <E: ProcessEdgesWork<VM=JikesRVM>> ScanBootImageRoots<E> {
+    pub fn new(subwork_id: usize, total_subworks: usize) -> Self {
```

Comment:
If I understand correct, those are worker_id and total_workers. The word subwork is confusing.

Reply:
There are a few other occurrences as well.

Reply:
`ScanBootImageRoots` is split into N sub-works (N=number of workers). But this does not enforce each worker thread to only execute one of these sub-works. If a worker finishes its sub-work early, it may get another sub-work from the global bucket. The scheduler imposes no restrictions on doing this. So I use `subwork_id` and `total_subworks` here instead of `worker_id` and `total_workers`.

Reply:
I see. Thanks for the explanation.